packages/components/src/border-box-control/test/index.js

```diff
@@ -277,8 +288,22 @@ describe( 'BorderBoxControl', () => {
 			);
 
 			const colorButtons = screen.getAllByLabelText( colorPickerRegex );
+			expect( colorButtons ).toHaveLength( 4 );
+
+			for ( let i = 0; i < colorButtons.length; i++ ) {
+				// Click on the color button to show the color picker popover.
+				await user.click( colorButtons[ 0 ] );
+
+				// Wait for color picker popover to fully appear
+				const colorPickerButton = screen.getByRole( 'button', {
+					name: 'Custom color picker.',
+				} );
+				await waitFor( () =>
+					expect(
+						getWrappingPopoverElement( colorPickerButton )
+					).toBePositionedPopover()
+				);
 
-			function assertStyleOptionsMissing() {
 				const styleLabel = screen.queryByText( 'Style' );
 				const solidButton = screen.queryByRole( 'button', {
```

Comment:
We should be able to shave ~8-10% of the test time by reducing the number of queries to look for style buttons. So, changing this:

```
				const solidButton = screen.queryByRole( 'button', {
					name: 'Solid',
				} );
				const dashedButton = screen.queryByRole( 'button', {
					name: 'Dashed',
				} );
				const dottedButton = screen.queryByRole( 'button', {
					name: 'Dotted',
				} );
				expect( styleLabel ).not.toBeInTheDocument();
				expect( solidButton ).not.toBeInTheDocument();
				expect( dashedButton ).not.toBeInTheDocument();
				expect( dottedButton ).not.toBeInTheDocument();
```

to:

```
				const styleButton = screen.queryByRole( 'button', {
					name: /(Solid)|(Dashed)|(Dotted)/,
				} );

				expect( styleButton ).not.toBeInTheDocument();
```

This exists in another test, so we could improve it there as well.